mobile/native/src/channel_fee.rs

```diff
@@ -0,0 +1,169 @@
+    pub fee: u32,
+}
+
+impl Subscriber for ChannelFeePaymentSubscriber {
+    fn notify(&self, event: &EventInternal) {
+        match event {
+            EventInternal::ChannelReady(channel_id) => {
+                self.register_funding_transaction(channel_id);
+            }
+            EventInternal::PaymentClaimed(amount_msats) => {
+                self.pay_funding_transaction_fees(*amount_msats);
+            }
+            _ => {}
+        }
+    }
+
+    fn events(&self) -> Vec<EventType> {
+        vec![EventType::ChannelReady, EventType::PaymentClaimed]
+    }
+}
+
+impl ChannelFeePaymentSubscriber {
+    pub fn new() -> Self {
+        Self {
+            open_channel_tx: Arc::new(Mutex::new(None)),
+        }
+    }
+
+    // Attempts to pay the transaction fees for opening an inbound channel.
+    fn pay_funding_transaction_fees(&self, amount_msats: u64) {
+        let transaction = match self.get_funding_transaction() {
+            Some(transaction) => transaction,
+            None => {
+                tracing::debug!("No pending funding transaction found!");
+                return;
+            }
+        };
+
+        tracing::debug!("Trying to pay channel opening fees of {}", transaction.fee);
+        let funding_tx_fees_msats = (transaction.fee * 1000) as u64;
+
+        if funding_tx_fees_msats > amount_msats {
+            tracing::warn!("Trying to pay fees with an amount smaller than the fees!")
+        }
+
+        let invoice = tokio::task::block_in_place(|| {
+            Handle::current().block_on(self.fetch_funding_transaction_fee_invoice(transaction.fee))
+        })
+        .ok();
+
+        if let Some(invoice) = invoice {
+            match ln_dlc::send_payment(&invoice) {
+                Ok(_) => {
+                    // unset the funding transaction marking it as being paid.
+                    self.unset_funding_transaction();
+                    tracing::info!("Successfully paid funding transaction fees of {funding_tx_fees_msats} msats to {}", config::get_coordinator_info().pubkey);
+                }
+                Err(e) => {
+                    tracing::error!("Failed to pay funding transaction fees of {funding_tx_fees_msats} msats to {}. Error: {e:#}", config::get_coordinator_info().pubkey);
+                }
+            }
+        }
+    }
+
+    // register jit channel opening transaction for fee payment
+    fn register_funding_transaction(&self, channel_id: &ChannelId) {
+        let channel_id_as_str = hex::encode(channel_id);
+        tracing::debug!("Received new inbound channel with id {channel_id_as_str}");
+
+        match ln_dlc::get_funding_transaction(channel_id) {
+            Ok(txid) => {
+                let transaction: Option<EsploraTransaction> = tokio::task::block_in_place(|| {
+                    Handle::current().block_on(self.fetch_funding_transaction(txid))
+                })
+                .ok();
+
+                if let Some(transaction) = transaction {
```

Comment:
🔧 I think we should emit an error if we can't find the transaction on the blockchain.